src/ow1/heroes/mei.opy

```diff
@@ -12,10 +17,49 @@ rule "[mei.opy]: Initialize Mei":
     initMei()
     eventPlayer.initialized = true
 
-rule "[mei.opy] Freeze enemy": # credit to snappycreeper for this rule
+rule "[mei.opy]: Track freezing on enemy":
     @Event playerDealtDamage
     @Hero mei
-    @Condition eventPlayer.isFiringPrimaryFire() == true
-    
-    wait(1.5, Wait.ABORT_WHEN_FALSE)
-    victim.setStatusEffect(eventPlayer, Status.FROZEN, 1.5)
+    @Condition eventPlayer.isFiringPrimaryFire() == true and victim.is_frozen == false
+
+    stopChasingVariable(victim.reset_time)
+    wait(0.016, Wait.ABORT_WHEN_FALSE) # wait 0.016 is 1 tick
+    if victim.start_freeze == false: # start the freeze time at 20
+        victim.freeze_percent = 20
+    if victim.freeze_percent < 100: # make sure it doesn't reset back to 20 
+        victim.start_freeze = true
+    victim.freeze_percent += 2.85 # 2.85% each tick from wiki
+    if victim.freeze_percent <= 70: # make sure you can only max freeze 70% movement speed
+        victim.setMoveSpeed(victim.freeze_percent)
+
+rule "[mei.opy]: Freeze enemy":
+    @Event playerDealtDamage
+    @Hero mei
+    @Condition eventPlayer.isFiringPrimaryFire() == true and victim.is_frozen == false
+
+    if victim.freeze_percent >= 100: # if its above 100, freeze em!
+        victim.setStatusEffect(eventPlayer, Status.FROZEN, 1.3)
+        victim.is_frozen = true
+        wait(1.5)
+        victim.setMoveSpeed(100)
+        victim.is_frozen = false
+        victim.start_freeze = false
+        victim.freeze_percent = 20
```

Comment:
```suggestion
    @Condition victim.freeze_percent >= 100: # if its above 100, freeze em!
    victim.setStatusEffect(eventPlayer, Status.FROZEN, 1.3)
    victim.is_frozen = true
    wait(1.5)
    victim.setMoveSpeed(100)
    victim.is_frozen = false
    victim.start_freeze = false
    victim.freeze_percent = 20
```

I think this is safe to do on the precondition/assumption that freeze_percent is calculated accurately. It's basically saying, you're frozen if you're 100% frozen.